packages/cozy-stack-client/src/JobCollection.js

```diff
@@ -55,6 +55,28 @@ class JobCollection {
     })
   }
 
+  /**
+   * Set the job's state
+   *
+   * @param {string} id: job's id
+   * @param {boolean} result: job's result
+   * @param {string} error: error message if any
+   *
+   * This does work only for jobs comming from @client triggers
+   */
+  async setState(id, result, error) {
+    return this.stackClient.fetchJSON('PATCH', `/jobs/${id}`, {
+      data: {
+        type: JOBS_DOCTYPE,
+        id,
+        attributes: {
+          state: result ? 'done' : 'errored',
+          error
+        }
+      }
+    })
+  }
```

Comment:
We should rely on the CozyClient DSL to be able to write: 

```jsx 
const existingJob = await client.query(Q('io.cozy.jobs').getById('1'));
const updatedJob = {
    ...existingJob,
    attributes: {
      ...existingJob.attributes,
      state: 'MyState',
      error: MyError
    }
}
client.save(updatedJob)
```

We should not create custom function in the collection, we should not bypass our CC mechanisms. Or at least, we should try to avoid it as much as we can. For a simple update, it's easy.

Here is the path: 
client.save => https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/CozyClient.js#L592-L596

getDocumentSavePlan => https://github.com/cozy/cozy-client/blob/6b59f8d6752d44afb8a10827af2cd2a9ca151d72/packages/cozy-client/src/CozyClient.js#L678

will make the call to create or update : https://github.com/cozy/cozy-client/blob/6b59f8d6752d44afb8a10827af2cd2a9ca151d72/packages/cozy-client/src/CozyClient.js#L684-L685 

updateDocument will send an `UPDATE_DOCUMENT` action: https://github.com/cozy/cozy-client/blob/90c7abb13fd71267cb44a1af6bca7b03b03a7645/packages/cozy-client/src/queries/dsl.js#L372

Links (in that case StackClient but can also be PouchLink) will call the update() method of the collection: https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/StackLink.js#L66-L69

To resume: 
This method should be called `update` and accept an io.cozy.jobs as argument.

Reply:
Thanks for the detailed comment